### PR TITLE
feature(knative): support configmap and secrets

### PR DESCRIPTION
**Pull request by lburgazzoli**
This add support for optionally mount configmaps and secrets to knative services, the default behaviour is still to use env var for backward compatibility.

@nicolaferraro do you mind trying this PR

### CODE REVIEW

File: docs/traits.adoc [hunk 1]
```diff
@@ -62,6 +62,22 @@ The following is a list of common traits that can be configured by the end users
...
+| Configure deployer behavior.
+  +
+  +
+  It's enabled by default.
+
+
+[cols="m,"]
+!===
+
+! deployer.container-image
```
Comment (nicolaferraro) [inline docs/traits.adoc]: Is this a "third" way of doing things?

File: pkg/util/util.go [hunk 2]
```diff
@@ -18,12 +18,15 @@ limitations under the License.
...
 
 import (
+	"encoding/base64"
 	"os"
 	"os/signal"
 	"path"
 	"regexp"
 	"syscall"
 
+	"github.com/magiconair/properties"
```
[the remark this answers is not in the document]
Comment (lburgazzoli) [reply pkg/util/util.go]: will do

File: docs/traits.adoc (hunk 1 above)
Comment (lburgazzoli) [reply docs/traits.adoc]: it is the only way now and it is used by both the deployment and the knative_service traits

**Comment by lburgazzoli** [comment]
@nicolaferraro ah damn yes, we can't use subPath 

Will rework

File: pkg/util/util.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lburgazzoli) [reply pkg/util/util.go]: done

**Comment by lburgazzoli** [comment]
should be better now

**Pull request merged**